src/bundle/Controller/LocationController.php

```diff
@@ -83,6 +85,26 @@ public function actionsAction(
         return $this->reloadTab('locations', $content->id, $redirectLocationId);
     }
 
+    public function swapLocationAction(
+        Content $content,
+        Location $location,
+        Request $request
+    ) {
+        $swapLocationsForm = $this->formFactory->createLocationsContentSwapForm();
+        $swapLocationsForm->handleRequest($request);
+
+        if ($swapLocationsForm->isValid()) {
+            $newLocationId = $swapLocationsForm->get('new_location_id')->getData();
+            $this->uiLocationService->swapLocations($location, $newLocationId);
+
+            return $this->resetLocation($content->id);
+        }
+
+        $redirectLocationId = $request->query->get('redirectLocationId', $content->contentInfo->mainLocationId);
+
+        return $this->reloadTab('locations', $content->id, $redirectLocationId);
```

Comment:
My bad. Okay, great :)

But it was specified by @SylvainGuittard that we should reset to the _same location id_, with the new content it has received. It kind of indicates that `resetLocation()` should be modified so that it can accept a Location, or location id (I'd prefer Location).

About it, actually, if it has to work with a `Content`, I'd create a distinct method, with a clear name like `resetToMainLocation($content)`.